### PR TITLE
Hv ivshmem barsize

### PR DESCRIPTION
**Pull request by chejianj**
Ivshmem spec says that BAR0 size should be 256 bytes. Windows ivshmem driver will check BAR0 size, if it is not 256 windows will refuse to load the driver. These two patches fixed this issue with the following
changes:

- In hv the BAR0 size is changed to 256 according to ivshmem spec.
  Reviewed-by: Fei Li <fei1.li@intel.com>
  Acked-by: Eddie Dong <eddie.dong@intel.com>

- In dm BAR base addr is aligned to 4K at least to satisfy EPT mapping.
  Reviewed-by: Fei Li <fei1.li@intel.com>
  Acked-by: Yu Wang <yu1.wang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

File: devicemodel/hw/pci/core.c [hunk 1]
```diff
@@ -564,7 +565,12 @@ pci_emul_alloc_resource(uint64_t *baseptr, uint64_t limit, uint64_t size,
 		return -1;
 	}
 
-	base = roundup2(*baseptr, size);
+	/* PCI spec said that BAR base should be naturally aligned. On ACRN
+	 * if the bar size < PAGE_SIZE, BAR base should be aligned with
+	 * PAGE_SIZE. This is because the minimal size that EPT can map/unmap
+	 * is PAGE_SIZE.
+	 */
```
Comment (dongyaozu) [inline devicemodel/hw/pci/core.c:572]: Should we simply replace base with PAGE_SIZE in this case?

Usually it means the space [size, PAGE_SIZE] is simply wasted.
If we use that space for other purpose, we may have trouble.

Comment (chejianj) [reply devicemodel/hw/pci/core.c:572]: Hi Eddie, 
Do you mean we just use PAGE_SIZE for this ivshmem case?

Yes, more MMIO address space will be wasted if we have many BARs whose size is < 4KB.
Currently all the BAR addresses are allocated uniformly in device model. 
When trying to allocate an address region for a new BAR, those gap used for alignment is just skipped and will not be used.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

File: devicemodel/hw/pci/core.c (hunk 1 above)
Comment (dongyaozu) [reply devicemodel/hw/pci/core.c:572]: I mean adjust size like following before Line 567:
 If (size < PAGE_SIZE)
    size = PAGE_SIZE;

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

File: devicemodel/hw/pci/core.c (hunk 1 above)
Comment (chejianj) [reply devicemodel/hw/pci/core.c:572]: Thanks for pointing this out.
Will submit a new PR.

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**